core/genesis.go

```diff
@@ -300,11 +300,20 @@ func ApplyOvmStateToState(statedb *state.StateDB, stateDump *dump.OvmDump, l1XDo
 	}
 	OVM_ETH, ok := stateDump.Accounts["OVM_ETH"]
 	if ok {
-		// Set the gateway of OVM_ETH
 		log.Info("Setting OVM_L1ETHGateway in OVM_ETH", "address", l1ETHGatewayAddress.Hex())
-		l1GatewaySlot := common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000000008")
-		l1GatewayValue := common.BytesToHash(l1ETHGatewayAddress.Bytes())
-		statedb.SetState(OVM_ETH.Address, l1GatewaySlot, l1GatewayValue)
+		if strings.Contains(OVM_ETH.Code, "a84ce98") {
```

Comment:
this is a methodID not present in the legacy contract

Reply:
Fine for now.